tests/s25Main/integration/testCheats.cpp

```diff
@@ -3,60 +3,94 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 
 #include "Cheats.h"
+#include "GamePlayer.h"
+#include "desktops/dskGameInterface.h"
 #include "worldFixtures/CreateEmptyWorld.h"
 #include "worldFixtures/WorldFixture.h"
+#include <turtle/mock.hpp>
 
 BOOST_AUTO_TEST_SUITE(CheatsTests)
 
 namespace {
-template<unsigned T_numPlayers>
-struct CheatsFixture : WorldFixture<CreateEmptyWorld, T_numPlayers>
+constexpr auto numPlayers = 1;
+constexpr auto worldWidth = 64;
+constexpr auto worldHeight = 64;
+struct CheatsFixture : WorldFixture<CreateEmptyWorld, numPlayers, worldWidth, worldHeight>
 {
-    Cheats cheats{this->world};
+    dskGameInterface gameDesktop{game, nullptr, 0, false};
+    Cheats& cheats = gameDesktop.GI_GetCheats();
+    const GameWorldViewer& viewer = gameDesktop.GetView().GetViewer();
+
+    GamePlayer& getPlayer(unsigned id) { return world.GetPlayer(id); }
+    GamePlayer& p1 = getPlayer(0);
+    const MapPoint p1HQPos = p1.GetHQPos();
 };
-using CheatsFixture1P = CheatsFixture<1>;
-using CheatsFixture2P = CheatsFixture<2>;
 } // namespace
 
-BOOST_FIXTURE_TEST_CASE(CheatsAreAllowed_InSinglePlayer, CheatsFixture1P)
-{
-    BOOST_TEST_REQUIRE(cheats.areCheatsAllowed() == true);
-}
-
-BOOST_FIXTURE_TEST_CASE(CheatsAreNotAllowed_InMultiplayer, CheatsFixture2P)
-{
-    BOOST_TEST_REQUIRE(cheats.areCheatsAllowed() == false);
-}
-
-BOOST_FIXTURE_TEST_CASE(CheatModeIsOffByDefault, CheatsFixture1P)
-{
-    BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == false);
-}
-
-BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOn, CheatsFixture1P)
+BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOnAndOffRepeatedly, CheatsFixture)
 {
+    BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == false); // initially off
     cheats.toggleCheatMode();
     BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == true);
-}
-
-BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOnAndOff, CheatsFixture1P)
-{
+    cheats.toggleCheatMode();
+    BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == false);
     cheats.toggleCheatMode();
     BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == true);
     cheats.toggleCheatMode();
     BOOST_TEST_REQUIRE(cheats.isCheatModeOn() == false);
 }
 
-BOOST_FIXTURE_TEST_CASE(CanToggleCheatModeOnAndOffRepeatedly, CheatsFixture1P)
+namespace {
+MOCK_BASE_CLASS(MockGameInterface, GameInterface)
+{
+    MOCK_METHOD(GI_PlayerDefeated, 1)
+    MOCK_METHOD(GI_UpdateMinimap, 1)
+    MOCK_METHOD(GI_FlagDestroyed, 1)
+    MOCK_METHOD(GI_TreatyOfAllianceChanged, 1)
+    MOCK_METHOD(GI_UpdateMapVisibility, 0)
+    MOCK_METHOD(GI_Winner, 1)
+    MOCK_METHOD(GI_TeamWinner, 1)
+    MOCK_METHOD(GI_StartRoadBuilding, 2)
+    MOCK_METHOD(GI_CancelRoadBuilding, 0)
+    MOCK_METHOD(GI_BuildRoad, 0)
+    MOCK_METHOD(GI_GetCheats, 0)
```

Comment:
`Unit tests / Linux (gcc-9, ubuntu-20.04, Debug, 3.16.3, 1.71.0) (pull_request)` gives some crap errors about `GI_GetCheats()` not being `noexcept`. I tried making it `noexcept` but I cannot mock a `noexcept` function properly with turtle. There are also some other warnings related to marking the mocked functions `override`, but I have no idea how to do that with turtle either.

Reply:
The override stuff can be ignored. However the noexcept issue is a long standing feature request for turtle.

My suggestion: Fix the noexcept warnings and to avoid having to mock the `GI_GetCheats` method just pass the Cheats instance in the ctor of the mock class and use a "regular" method.